state/indexer/block/kv/kv.go

```diff
@@ -199,10 +201,13 @@ func (idx *BlockerIndexer) Search(ctx context.Context, q *query.Query) ([]int64,
 
 		select {
 		case <-ctx.Done():
-			break
+			cont = false
```

Comment:
wait, I'm missing something here. What does breaking `if !cont` change as opposed to simply breaking here?

Reply:
`break` in the original code will break out of the `select` (i.e. do literally nothing), instead of the `for`.